### PR TITLE
 feat: add gv-cron-editor component

### PR DESCRIPTION
**Pull request by gcusnieux**
fix gravitee-io/issues#4043

### CODE REVIEW

**Review by gaetanmaisse — DISMISSED** [review]

**Comment by gaetanmaisse** [comment]
It looks like the `gv-cron-editor` stories are sometimes freezing 🤔 

Storybook keeps loading and I can't interact with the editor. Everything is working fine in the story with input values.

Can someone else try to access to: http://localhost:6006/?path=/story/molecules-gv-cron-editor--simple

**Comment by gcusnieux** [comment]
@gaetanmaisse yes.. I have the problem and i don't understand why 😢 
It's useful to have everything in one story...  Maybe this issue #238 could help?

**Comment by gaetanmaisse** [comment]
Yep 🤞🏻 I will try to 🔍 a bit what can be done easily, in the meantime, this PR is ✅  for me

File: src/atoms/gv-date-picker-calendar.js [hunk 1]
```diff
@@ -312,26 +332,29 @@ export class GvDatePickerCalendar extends LitElement {
...
   render () {
     const classes = {
       'calendar-grid': true,
       'calendar-grid-large': this.monthMode || this.yearMode,
     };
     return html`
     <div class="box">
-      <div class="box-col">
-          <div class="nav">
+      ${this.time === true && this.strict === true ? '' : html`<div class="box-col">
```
Comment (ytvnr) [inline src/atoms/gv-date-picker-calendar.js:346]: maybe you can use `isStrictTime()` ?

File: src/atoms/gv-input.js [hunk 2]
```diff
@@ -298,11 +298,13 @@ export class GvInput extends InputElement(LitElement) {
 
       setTimeout(() => {
         const clipboardPopover = this.shadowRoot.querySelector('.clipboard__popover');
-        this.getInputElement().addEventListener('click', (e) => {
-          e.preventDefault();
-          e.stopPropagation();
-          clipboardPopover.click();
-        });
+        if (this.readonly) {
```
Comment (ytvnr) [inline src/atoms/gv-input.js:301]: don't know if it is related but I think there is a glitch on readonly story -> the value filled are not in the good format so the controls appears as invalid

image

Comment (ytvnr) [inline src/atoms/gv-input.js:301]: don't know if it is related but I think there is a glitch on readonly story -> the value filled are not in the good format so the controls appears as invalid

image

File: src/atoms/gv-input.js [hunk 3]
```diff
@@ -351,6 +359,27 @@ export class GvInput extends InputElement(LitElement) {
...
+      let key;
+      if (e.type === 'paste') {
+        key = e.clipboardData.getData('text/plain');
+      }
+      else {
+        key = String.fromCharCode(e.keyCode);
+      }
+
+      let regex = /[0-9]/;
+      if ((this.min == null || this.min < 0)) {
```
Comment (ytvnr) [inline src/atoms/gv-input.js:373]:
```suggestion
      if (this.min == null || this.min < 0) {
```

File: src/molecules/gv-cron-editor.js [hunk 4]
```diff
@@ -0,0 +1,615 @@
...
+  constructor () {
+    super();
+    this._value = '';
+    this.readonly = false;
+    this.autofocus = false;
+    this.clipboard = false;
+    this._modes = ['seconds', 'minutes', 'hourly', 'daily', 'weekly', 'monthly', 'yearly', 'pro'];
+    this._timeRange = [...Array(60).keys()];
+    this._days = [
+      { value: 'MON', label: 'Monday' },
```
Comment (ytvnr) [inline src/molecules/gv-cron-editor.js:66]: do you think these label should be translated ?

**Review by ytvnr — COMMENTED** [review]
some remarks and questions bug great work !!

File: src/molecules/gv-cron-editor.js (hunk 4 above)
Comment (gcusnieux) [reply src/molecules/gv-cron-editor.js:66]: @aelamrani @phiz71 What do you think ?

Comment (aelamrani) [reply src/molecules/gv-cron-editor.js:66]: not useful, it is an admin component

File: src/atoms/gv-input.js (hunk 2 above)
Comment (gcusnieux) [reply src/atoms/gv-input.js:301]: Well seen, it was a mistake... 
A readonly field can't be invalid even with the wrong value

Comment (gcusnieux) [reply src/atoms/gv-input.js:301]: ✅

File: src/atoms/gv-date-picker-calendar.js (hunk 1 above)
Comment (gcusnieux) [reply src/atoms/gv-date-picker-calendar.js]: ✅

**Review by ytvnr — APPROVED** [review]
Good job !

**Pull request merged**